File: android/src/main/kotlin/dev/hebei/mars_logging/MarsLoggingPlugin.kt
package dev.hebei.mars_logging

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** MarsLoggingPlugin */
class MarsLoggingPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        val api = XLogImpl()
        XLogApi.setUp(binding.binaryMessenger, api)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        XLogApi.setUp(binding.binaryMessenger, null)
    }
}

File: macos/Classes/MarsLoggingPlugin.swift
import Cocoa
import FlutterMacOS

public class MarsLoggingPlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let messenger = registrar.messenger
        let api = XLogImpl()
        XLogApiSetup.setUp(binaryMessenger: messenger, api: api)
    }
}
